### PR TITLE
feat(client-wrapper): Adds support for CVR

### LINKED ISSUE #47: Add support for cStorVolume CR
Add support to list, get and delete the cstorVolume(CVR) CR which is a clustered-scope resource.

you have to use the package `openebsv1aplha1 "github.com/openebs/CITF/pkg/apis/openebs.io/v1alpha1"`

### PR DESCRIPTION
**Pull request by bharathpgp**
Fixes #47

Signed-off-by: bharathpgp <bharathpgp@gmail.com>

### CODE REVIEW

File: utils/k8s/openebscrd.go [hunk 1]
```diff
@@ -40,3 +40,26 @@ func (k8s K8S) DeleteStoragePoolClaim(spcName string) error {
...
+
+// GetCStorVolumeReplica returns the CStorVolumeReplica object for given cvrName, namespace.
+// :return: *openebs_v1.CStorVolumeReplica: Pointer to CStorVolumeReplica objects.
+func (k8s K8S) GetCStorVolumeReplica(cvrName, namespace string) (*openebs_v1.CStorVolumeReplica, error) {
+	cvrClient := k8s.OpenebsClientSet.Openebs().CStorVolumeReplicas(namespace)
+	return cvrClient.Get(cvrName, meta_v1.GetOptions{})
+}
+
+// ListCStorVolumeReplica returns all the CStorVolumeReplica objects for given namespace
+func (k8s K8S) ListCStorVolumeReplica(namespace string) ([]openebs_v1.CStorVolumeReplica, error) {
```
Comment (princerachit) [inline utils/k8s/openebscrd.go]: Can this be modified to return an encapsulating object like `CStorVolumeReplicaList` I see some other PRs are using this way.

File: utils/k8s/openebscrd.go [hunk 2]
```diff
@@ -40,3 +40,22 @@ func (k8s K8S) DeleteStoragePoolClaim(spcName string) error {
 	spcClient := k8s.OpenebsClientSet.Openebs().StoragePoolClaims()
 	return spcClient.Delete(spcName, &meta_v1.DeleteOptions{})
 }
+
+// GetCStorVolumeReplica returns the CStorVolumeReplica object for given cvrName, namespace.
+// :return: *openebs_v1.CStorVolumeReplica: Pointer to CStorVolumeReplica objects.
```
Comment (sonasingh46) [inline utils/k8s/openebscrd.go]: We might not need line 45 comment.

File: utils/k8s/openebscrd.go [hunk 3]
```diff
@@ -40,3 +40,22 @@ func (k8s K8S) DeleteStoragePoolClaim(spcName string) error {
...
+	return cvrClient.Get(cvrName, meta_v1.GetOptions{})
+}
+
+// ListCStorVolumeReplica returns all the CStorVolumeReplicaList for given namespace
+func (k8s K8S) ListCStorVolumeReplica(namespace string) (*openebs_v1.CStorVolumeReplicaList, error) {
+	cvrClient := k8s.OpenebsClientSet.OpenebsV1alpha1().CStorVolumeReplicas(namespace)
+	return cvrClient.List(meta_v1.ListOptions{})
+}
+
+// DeleteCStorVolumeReplica deletes a CStorVolumeReplica with the given cvrName, namespace
```
Comment (Akash4927) [inline utils/k8s/openebscrd.go]: It should be `with the given cvrName and namespace`

File: utils/k8s/openebscrd.go [hunk 4]
```diff
@@ -40,3 +40,21 @@ func (k8s K8S) DeleteStoragePoolClaim(spcName string) error {
 	spcClient := k8s.OpenebsClientSet.Openebs().StoragePoolClaims()
 	return spcClient.Delete(spcName, &meta_v1.DeleteOptions{})
 }
+
+// GetCStorVolumeReplica returns the CStorVolumeReplica object for given cvrName and namespace
+func (k8s K8S) GetCStorVolumeReplica(cvrName, namespace string) (*openebs_v1.CStorVolumeReplica, error) {
+	cvrClient := k8s.OpenebsClientSet.OpenebsV1alpha1().CStorVolumeReplicas(namespace)
+	return cvrClient.Get(cvrName, meta_v1.GetOptions{})
```
Comment (sonasingh46) [inline utils/k8s/openebscrd.go]: Receive this ` meta_v1.GetOptions{}`  as an argument.

File: utils/k8s/openebscrd.go [hunk 5]
```diff
@@ -40,3 +40,21 @@ func (k8s K8S) DeleteStoragePoolClaim(spcName string) error {
...
+}
+
+// ListCStorVolumeReplica returns all the CStorVolumeReplicaList for given namespace
+func (k8s K8S) ListCStorVolumeReplica(namespace string) (*openebs_v1.CStorVolumeReplicaList, error) {
+	cvrClient := k8s.OpenebsClientSet.OpenebsV1alpha1().CStorVolumeReplicas(namespace)
+	return cvrClient.List(meta_v1.ListOptions{})
```
Comment (sonasingh46) [inline utils/k8s/openebscrd.go]: Receive this `meta_v1.ListOptions{}` as an argument.

File: utils/k8s/openebscrd.go [hunk 6]
```diff
@@ -40,3 +40,21 @@ func (k8s K8S) DeleteStoragePoolClaim(spcName string) error {
...
+}
+
+// DeleteCStorVolumeReplica deletes a CStorVolumeReplica with the given cvrName and namespace
+func (k8s K8S) DeleteCStorVolumeReplica(cvrName, namespace string) error {
+	cvrClient := k8s.OpenebsClientSet.OpenebsV1alpha1().CStorVolumeReplicas(namespace)
+	return cvrClient.Delete(cvrName, &meta_v1.DeleteOptions{})
```
Comment (sonasingh46) [inline utils/k8s/openebscrd.go]: Receive `&meta_v1.DeleteOptions{}` as an argument.

**Comment by sonasingh46** [comment]
As well please resolve the conflict.

**Pull request merged**